tao_compiler/mlir/disc/transforms/mhlo_mark_shape_calc.cc

```diff
@@ -90,6 +90,21 @@ void DiscMarkShapeCalc::MarkShapeCalcOps() {
   }
 }
 
+// NOTE(lanbo.llb): mhlo.disc op will produce 2 outputs, the 2nd output is i64
+// tensor with single element represent number of valid elements in the 1st
+// output, this will be used to calc the final output shape. It is indeed a
+// shape calc, but where op should not be considered as shape calc op. By
+// marking it as shape calc op will cause mhlo_disc.where impossible to fuse in
+// later fusion pass.
+bool withInputFromWhereOp(Operation* op) {
+  for (Value operand : op->getOperands()) {
+    if (isa<mhlo_disc::WhereOp>(operand.getDefiningOp())) {
```

Comment:
If the operand is the first operand of where op, should we also need to treat it specially?

Reply:
Currently based on the computation pattern we have observed, we want to treat `mhlo_disc.where` as compute op, thus we do not treat them differently.